hw/syn/tools/dc/sweep.tcl

```diff
@@ -0,0 +1,115 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Simple tcl script for DC to do some wire-load-model-based sweep syntheses.
+
+#####################
+##  PREPARE FLOW   ##
+#####################
+
+source ../../../foundry/syn/dc/setup.tcl
+
+# paths
+set WORKLIB  "WORK"
+set REPDIR   "REPORTS"
+set DDCDIR   "DDC"
+set VLOGDIR  "NETLISTS"
+
+exec mkdir -p ${REPDIR} ${DDCDIR} ${VLOGDIR} ${WORKLIB}
+
+# define work lib path
+define_design_lib WORK -path $WORKLIB
+
+######################
+##   DESIGN SETUP  ##
+######################
+
+set DUT "prim_prince"
```

Comment:
is this an artifact?

Reply:
hey @cdgori, not this is only semi-generic, meaning that it should serve as a template for designers to "hack around" and do some local test syntheses. the clean and generic way for complete comportable IPs is to run the flow integrated into the dvsim tool. so this here is just an example, using `prim_prince`.